File: Common/python/ListNode.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
        
    def to_list(self):
        list = []
        current = self
        while current:
            list.append(current.val)
            current = current.next
        return list

    @staticmethod
    def create_linked_list(values):
        if not values:
            return None
        head = ListNode(values[0])
        current = head
        for value in values[1:]:
            current.next = ListNode(value)
            current = current.next
        return head

File: Common/java/ListNode.java
package Common.java;

import java.util.List;

// Definition for singly-linked list.
public class ListNode {
  public int val;
  public ListNode next;
  public ListNode() {}
  public ListNode(int val) { this.val = val; }
  public ListNode(int val, ListNode next) { this.val = val; this.next = next; }
  public ListNode(List<Integer> list) {
    this(list.size() > 0 ? list.get(0) : -1, list.size() > 1 ? new ListNode(list.subList(1, list.size())) : null);
  }

  @Override
  public String toString() {
    return "" + val + (next == null ? "" : "," + next.toString());
  }
}
